packages/build-utils/src/file-ref.ts

```diff
@@ -1,10 +1,20 @@
 import assert from 'assert';
-import fetch from 'node-fetch';
 import multiStream from 'multistream';
 import retry from 'async-retry';
 import Sema from 'async-sema';
 import { FileBase } from './types';
 
+// Necessary for `[vercel dev] Should work with nested `tsconfig.json` files`
+// to succeed
+let fetch: Awaited<typeof import('node-fetch').default> | undefined;
+async function initFetch() {
+  if (fetch !== undefined) {
+    return;
+  }
+  const nodeFetch = await import('node-fetch');
+  fetch = nodeFetch.default;
+}
```

Comment:
Not sure why dynamically importing `node-fetch` fixes the behavior. Curious if anyone knows why this could be, and I'm open to other ideas.

Though, this seems like the least invasive solution to removing the deprecation warning without failing tests!

Reply:
I'm glad this solves the problem, but I really want to understand why

Reply:
This is the error that happens without the dynamic require, right?

```
Error: Cannot find module '/var/task/.next/server/app/(newroot)/nested-dynamic-params-group/params-false/[slug]/page.js'
Require stack:
- /var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js
- /var/task/___next_launcher.cjs
- /opt/rust/nodejs.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
    at /var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:19:44583
    at Module._load (node:internal/modules/cjs/loader:1051:27)
    at /opt/rust/nodejs.js:1:11700
    at Function.ir (/opt/rust/nodejs.js:1:12074)
    at e.<computed>.te._load (/opt/rust/nodejs.js:1:11670)
    at Module.require (node:internal/modules/cjs/loader:1311:19)
    at w.require (/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:19:44792)
    at require (node:internal/modules/helpers:179:18)
    at ni (/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js:33:12279) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/var/task/node_modules/next/dist/compiled/next-server/server.runtime.prod.js',
    '/var/task/___next_launcher.cjs',
    '/opt/rust/nodejs.js'
  ],
  page: '/nested-dynamic-params-group/params-false/second'
}
```

Reply:
Not sure what that error is. When visiting `/api` with that dev server locally I see:
```
> Ready! Available at http://localhost:3000
 ✓ Ready in 906ms
/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:7677
    var AsyncIteratorPrototype = Object.getPrototypeOf(Object.getPrototypeOf(async function* () {
                                        ^
TypeError: Cannot convert undefined or null to object
    at Function.getPrototypeOf (<anonymous>)
    at ../../node_modules/.pnpm/whatwg-url@12.0.1/node_modules/whatwg-url/lib/utils.js (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:7677:41)
    at __require (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:9:50)
    at ../../node_modules/.pnpm/whatwg-url@12.0.1/node_modules/whatwg-url/lib/URL.js (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:10386:17)
    at __require (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:9:50)
    at ../../node_modules/.pnpm/whatwg-url@12.0.1/node_modules/whatwg-url/webidl2js-wrapper.js (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:10732:15)
    at __require (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:9:50)
    at ../../node_modules/.pnpm/whatwg-url@12.0.1/node_modules/whatwg-url/index.js (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:10743:36)
    at __require (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:9:50)
    at ../../node_modules/.pnpm/node-fetch@2.6.7/node_modules/node-fetch/lib/index.js (/Users/onsclom/projects/vercel/packages/build-utils/dist/index.js:10778:37)
Error: Function `api/index.ts` failed with exit code 1
```

Reply:
Specifically I'm doing `pnpm dev dev --cwd test/dev/fixtures/nested-tsconfig` and then visiting http://localhost:3000/api to see this error.